src/coreclr/src/vm/eepolicy.cpp

```diff
@@ -1056,7 +1149,26 @@ void DECLSPEC_NORETURN EEPolicy::HandleFatalStackOverflow(EXCEPTION_POINTERS *pE
 
     STRESS_LOG0(LF_EH, LL_INFO100, "In EEPolicy::HandleFatalStackOverflow\n");
 
-    DisplayStackOverflowException();
+    FrameWithCookie<FaultingExceptionFrame> fef;
+#if defined(FEATURE_EH_FUNCLETS)
+    *((&fef)->GetGSCookiePtr()) = GetProcessGSCookie();
+#endif // FEATURE_EH_FUNCLETS
+    if (pExceptionInfo && pExceptionInfo->ContextRecord)
+    {
+        GCX_COOP();
+        AdjustContextForWriteBarrier(pExceptionInfo->ExceptionRecord, pExceptionInfo->ContextRecord);
+        fef.InitAndLink(pExceptionInfo->ContextRecord);
+    }
+
+    static volatile LONG g_stackOverflowCallStackLogged = 0;
+
+    // Dump stack trace only for the first thread failing with stack overflow to prevent mixing
```

Comment:
Are the other threads going to wait for this one to finish printing?

Reply:
Yes, I've verified that. I've tried to put a 10 second sleep after printing the stack trace and it was still printed while the other threads were passing by.

Reply:
Even if the other threads hit stackoverflow exception as well? It may be useful to note in the comment where the other threads are going to wait.

Reply:
I was testing it with many threads hitting stack overflow at the same time. But I've just found that the threads wait only when running under a debugger. When running without a debugger, nothing is waiting for this thread to complete and so we get partial or even no stack trace dump here. So I'll need to fix it.